client/src/components/signin.js

```diff
@@ -18,8 +18,10 @@ export class SignIn extends React.Component {
 
     event.preventDefault();
     // console.log('my identifyUser', this)
-    this.props.identifyUser({email: this.email, password: this.password});
-    browserHistory.push('/podcast-list');
+    this.props.identifyUser({email: this.email, password: this.password}, () => {
```

Comment:
Callback magic! Only get redirected if you login successfully. The only problem I see here is there is no way of informing the user visually when they make mistakes such as the user not being in the database, or incorrect username/passwords being used.